src/app/blog/[slug]/page.tsx

```
@@ -0,0 +1,216 @@
+    notFound();
+  }
+
+  const relatedPosts = getRelatedPosts(post.slug, 3);
+  const structuredData = generateStructuredData(post);
+
+  return (
+    <>
+      <Script
+        id="structured-data"
+        type="application/ld+json"
+        dangerouslySetInnerHTML={{ __html: JSON.stringify(structuredData) }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)